source/standards/source-code.html.md.erb

```diff
@@ -64,6 +64,31 @@ If an application is no longer used in production, you should [archive its repos
 
 Update the application's README to explain why the repository has been archived, and link to a new location if the application has been superseded.
 
+### Using Github Actions and workflows
+
+Alphagov repositories can be configured to use [GitHub Actions](https://docs.github.com/en/actions) for CI/CD jobs, for example running unit tests or deploying a static site.
+
+Actions and workflows can be powerful, so take care to follow best security practice.
+
+Ensure the repository permissions follow [the principle of least privilege](/standards/principle-least-access.html).
+Set workflow permissions to read-only by default, and disable all actions on repositories that are not using them (this will hide the 'Actions' tab from the repository menu).
+
+If your repository has external contributors, [ensure they do not have permissions to add workflows or trigger workflow runs](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks).
+
+If your workflow interacts with another resource (for example AWS or DockerHub), consider setting up a dedicated account or role, with permissions limited to the scope of the action.
```

Comment:
Maybe work for a future PR, but I'd like to see some discussion of how to store secrets.

When should they be stored at the repository level? When should they be stored at the organisation level?

If you have lots of repositories that access the same resource (say DockerHub), presumably you should avoid duplicating the credentials for that service across all the repositories (as that would be a headache if you needed to rotate the token). Org level secrets seem to solve this problem (you can scope them to specific repos, I think). But I'm not sure how much GitHub permission you need to set those - only org owners perhaps?

On the subject of org owners, is it worth mentioning that org owners from other teams in alphagov may be able to extract your credentials?

Reply:
@richardTowers yeah, I had a quick look at org secrets and it says:
> Only alphagov administrators may add organization secrets.

Reply:
@richardTowers out of interest do you know how other org owners may extract credentials. It seems from my limited testing that repo level secrets are not viewable by others once set i.e. similar to AWS secret keys in that way, although I suppose someone with permission to alter a workflow could extract a secret by executing a job.

Reply:
Yep - secrets and tokens are a whole 'nother can of worms (agree it might be better in another PR). 

That reminds me of another permissions wrinkle I forgot to mention: Pay were looking at using a workflow to create a tag. To do that, we would have needed to change the workflow permissions on the repo to [`write: contents`](https://docs.github.com/en/rest/reference/permissions-required-for-github-apps#permission-on-contents), which basically would grant any third party action in that workflow a whole bunch of extra permissions 🙀 This is [a long-running issue with Github's permission model](https://github.com/dear-github/dear-github/issues/113), but if someone was using a potentially dodgy third party action, the attack surface would be much bigger.

Reply:
@gidsg - yeah, that's my understanding too. I was thinking of an org owner modifying a workflow, or exposing an org level secret to another repo under their control and then echo'ing it in their own workflow. I don't think they could do that without leaving some audit trail, but they could probably do it without being spotted.